test_conformance/conversions/conversions_data_info.h

```diff
@@ -717,6 +726,33 @@ void DataInfoSpec<InType, OutType, InFP, OutFP>::conv_sat(OutType *out,
     }
 }
 
+template <typename InType, typename OutType, bool InFP, bool OutFP>
+void DataInfoSpec<InType, OutType, InFP, OutFP>::set_allow_zero(uint8_t *allow,
+                                                                OutType *out,
+                                                                InType *in)
+{
+  // from double
+  if (std::is_same<InType, cl_double>::value)
+    *allow |= IsDoubleSubnormal(*in);
+  // from float
+  if (std::is_same<InType, cl_float>::value)
+    *allow |= IsFloatSubnormal(*in);
+  // from half
+  if (is_in_half())
+    *allow |= IsHalfSubnormal(*in);
+
+  // handle the cases that the converted result is subnormal
+  // from double
+  if (std::is_same<OutType, cl_double>::value)
+    *allow |= IsDoubleSubnormal(*out);
+  // from float
+  if (std::is_same<OutType, cl_float>::value)
+    *allow |= IsFloatSubnormal(*out);
+  // from half
+  if (is_out_half())
+    *allow |= IsHalfSubnormal(*out);
```

Comment:
Checking: Previously, we were only checking the source data, which is now called `in`.  This section of code is also checking the destination data also, called `out`.  Do we need to check both?

Reply:
It's necessary to check the destination data, called out, in class template member function DataInfoSpec<InType, OutType, InFP, OutFP>::set_allow_zero to address the case when the destination data is a denormalized number before rounding, and the execution platform doesn’t support denormalized numbers in destination data type.

Reply:
Gotcha, thanks!